### ISSUE TITLE
[Databricks] Same job name name for different job

### ISSUE THREAD

**Issue by dkt-sophie-ly** [OpenLineage/OpenLineage#3143]
Hi all ! 

Not sure if it's a real issue but we integrated OpenLineage in our databricks cluster and a lot of users use notebook to read data (without writing any).
So a lot of job have the same name like: cluster_id.cloud.databricks.com.adaptive_spark_plan
For this kind of job it creates a lot of different 'runs' but it's actually different notebook created by different user. 

We try to do some post-processing of the event by adding the notebook name (we have the information in run → facets → environment-properties → environment-properties → spark → databricks → notebook → path

But this facet is only available in a RUNNING event (not in START and COMPLETE) so that makes kinda difficult to do some post-processing. 

Do you think it's an integration issue ? Can we add these facets in the START and COMPLETE event ? 

Thanks for your reply !

**Comment by dkt-sophie-ly** [comment]
I'll git it a try, thanks for your response !

**Comment by algorithmy1** [comment]
Hello @pawel-big-lebowski ,

Aside from having the same facet in the three events, what do you think about how we're constructing the job names? I feel like when a job doesn't write any data, many jobs end up with the same name. In certain cases, do you think it would make sense to add some additional information to differentiate them? For example, we could include the source code file name or the notebook ID, etc.

Let me know your thoughts.

**Comment by pawel-big-lebowski** [comment]
@algorithmy1 you're right, the job name identifier should be unique. That's why we are adding output dataset name as a job suffix. Because otherwise job name is assigned based on Spark action's name and these name are the same for all `collect`, `show` and other actions. Apart from uniqueness, job name should be reproducible, so if you open same databricks notebook and run the same cell another day, this should be the same OpenLineage job I think. And the last requirement for job names is to make them human readable, so that person looking at job name gets the feeling of what is this job about.

I am not sure how to meet all the criteria above for databricks notebooks that don't write data to any datasets. But adding notebook ID would be definitely an improvement and one can implement it optional (turned on/off though configuration) in case someone in future has other solution to this.
